src/ethereum/wallet.rs

```diff
@@ -290,28 +308,43 @@ impl Wallet {
             .send_raw_transaction(transaction_hex)
             .await?;
 
-        // TODO: Return TransactionReceipt instead
-        std::thread::sleep(std::time::Duration::from_millis(2000));
+        let _ = self.wait_until_transaction_receipt(hash).await?;
 
         Ok(hash)
     }
 
     pub async fn get_transaction_by_hash(
         &self,
         transaction_hash: Hash,
-    ) -> anyhow::Result<comit::ethereum::Transaction> {
+    ) -> anyhow::Result<Transaction> {
         self.geth_client
             .get_transaction_by_hash(transaction_hash)
             .await
     }
 
-    pub async fn get_transaction_receipt(
+    pub async fn wait_until_transaction_receipt(
         &self,
         transaction_hash: Hash,
     ) -> anyhow::Result<TransactionReceipt> {
-        self.geth_client
-            .get_transaction_receipt(transaction_hash)
-            .await
+        let start_time = std::time::Instant::now();
+        let max_retry_time = Duration::from_millis(60_000);
```

Comment:
❓ Is maximum retry time arbitrarily chosen?

Reply:
Yep. I thought about making it configurable, but at this stage it seems excessive. Maybe we don't need a `max_retry_time` at all though. I just want to prevent this waiting forever, but maybe that's also over-engineered.